### DISCUSSION TITLE
回收站可以预留一定的安全空间，当回收持续写入剩余空间小于这个安全空间后，不支持回收。

### DISCUSSION THREAD

**Discussion by JAINKRE** [1Panel-dev/1Panel#5596] [Q&A]
### 1Panel 版本

v1.9.4

### 问题描述

* 环境centos 7.9

对文件夹执行压缩（zip压缩），会在根目录`.1panel_clash`文件夹中产生缓存，且该缓存不会执行完毕后自动删除，当压缩多个大文件时，发现会持续写入缓存，直至写满，导致centos错误。

以下是机器盘占用信息：
![20240118-171612](https://github.com/1Panel-dev/1Panel/assets/63288007/be88439b-8578-4b55-804a-bd4cc2025e5f)

可以看到`/dev/mapper/centos-root`空间被写满

以下是执行压缩后`.1panel_clash`文件夹中缓存：
![20240118-172054](https://github.com/1Panel-dev/1Panel/assets/63288007/7fd4b92a-ea8f-49e4-9157-428dc4fddd09)

一旦出现这种情况，系统所有服务全部不可用。必须手动登录服务器，删除部分文件夹后，才可以正常登录。

### 重现步骤

* 对任意大文件夹执行zip压缩
* 根目录`.1panel_clash`文件夹产生缓存
* centos安装时默认root路径空间仅50GB，当写满后服务器无法继续读写文件，会导致系统错误

### 期待的正确结果

建议以下解决办法：
1. 添加缓存删除机制
2. 压缩缓存建议写在压缩对象所在同级目录下，不建议写至根目录，因为当服务器安装多块硬盘时，比如系统盘为ssd固态，而执行压缩的路径挂载在m.2固态中，会导致写入性能降低
3. 优化压缩策略，不使用缓存

**Comment by zhengkunwang223** [comment]
.1panel_clash 这个目录是回收站目录 当你删除文件的时候会默认把文件放在这个目录 方便你找回误删的文件
zip 压缩正常情况不会在此文件增加缓存 你可以把回收站清空 然后再压缩 观察此文件夹是否会增加文件或者存储占用

**Comment by JAINKRE** [comment]
好的，问题已经排查清除，确实和压缩无关，感谢说明。但还是建议优化回收机制，因为当持续向回收目录写入文件后，还是会导致root路径空间占用激增，从而影响系统使用，而且一旦空间占满后，即使登录主机也没办法查询和删除`.1panel_clash`文件夹内容

比如可以预留一定的安全空间，当回收持续写入剩余空间小于这个安全空间后，不支持回收。

**Comment by zhengkunwang223** [comment]
> 好的，问题已经排查清除，确实和压缩无关，感谢说明。但还是建议优化回收机制，因为当持续向回收目录写入文件后，还是会导致root路径空间占用激增，从而影响系统使用，而且一旦空间占满后，即使登录主机也没办法查询和删除`.1panel_clash`文件夹内容
> 
> 比如可以预留一定的安全空间，当回收持续写入剩余空间小于这个安全空间后，不支持回收。

回收不会导致你磁盘使用率增加 回收只是把文件换了一个位置 其实这个问题的根本问题是 磁盘使用率高了之后需要提醒用户 我们后续版本会优化此处
